insights/mappers/xfs_info.py

```diff
@@ -47,19 +109,46 @@ def parse_content(self, content):
                     sect_info['specifier'] = match.group('specifier')
                     if match.group('specval'):
                         sect_info['specifier_value'] = match.group('specval')
-                for pair in self.keyval_re.findall(match.group('keyvaldata')):
-                    (key, value) = pair
+                for key, value in keyval_re.findall(match.group('keyvaldata')):
                     if value[-1] != 's':
                         # Value doesn't end with 'blks', so convert it to int.
                         value = int(value)
                     sect_info[key] = value
 
         self.xfs_info = xfs_info
 
+        # Check that the meta-data and data sections are there and contain
+        # the things we expect
+        if 'meta-data' not in xfs_info:
+            raise ParseException("No 'meta-data' section found")
+        if 'specifier' not in xfs_info['meta-data']:
+            raise ParseException("Device specifier not found in meta-data")
+        if 'data' not in xfs_info:
+            raise ParseException("No 'data' section found")
+        if 'blocks' not in xfs_info['data']:
+            raise ParseException("'blocks' not defined in data section")
+        if 'bsize' not in xfs_info['data']:
+            raise ParseException("'bsize' not defined in data section")
+        if 'log' not in xfs_info:
+            raise ParseException("No 'log' section found")
+        if 'blocks' not in xfs_info['log']:
+            raise ParseException("'blocks' not defined in log section")
+        if 'bsize' not in xfs_info['log']:
+            raise ParseException("'bsize' not defined in log section")
+
+        # Store a few handy properties
+        self.device = xfs_info['meta-data']['specifier']
+
         # Calculate a few things
         self.data_size = xfs_info['data']['blocks'] * xfs_info['data']['bsize']
         self.log_size = xfs_info['log']['blocks'] * xfs_info['log']['bsize']
 
+        # Determine mount from file name if possible:
+        mount_name_re = re.compile('xfs_info_(?P<mount>.*)')
+        match = mount_name_re.search(self.file_name)
+        if match:
+            # Try to put slashes back...
+            self.mount = match.group('mount').replace('.', '/')
```

Comment:
`self.mount` should have a documented default value so that it will be present in the object regardless of whether the mount information is parsed.